nightfall-deployer/contracts/Challenges.sol

```diff
@@ -138,20 +154,70 @@ contract Challenges is Stateful, Key_Registry, Config {
     );
     // check the historic root is in the block provided.
     require(
-      transactions[transactionIndex].historicRootBlockNumberL2 ==
+      transactions[transactionIndex].historicRootBlockNumberL2[0] ==
       blockNumberL2ContainingHistoricRoot,
       'Incorrect historic root block'
     );
     ChallengesUtil.libChallengePublicInputHash(
       transactions[transactionIndex],
-      blockL2ContainingHistoricRoot.root
+      [blockL2ContainingHistoricRoot.root, ZERO]
     );
     // Delete the latest block of the two
     challengeAccepted(blockL2, blockNumberL2);
   }
 
-  // This function signature is used when we have a zero historic root
-  // i.e. a deposit transaction.
+  /**
+  This checks if the public input hash is incorrect, as the public input hash
+  is calculated differently for each transaction type, we overload this function.
+
+  This function signature is used when we have two historic root
+  i.e. a double transfer. We need to verify two historic blocks.
+  */
+  function challengePublicInputHash(
+    Block memory blockL2,
+    uint256 blockNumberL2,
+    Transaction[] memory transactions,
+    uint256 transactionIndex,
+    Block memory blockL2ContainingHistoricRoot,
+    Block memory blockL2ContainingHistoricRoot2,
+    uint256 blockNumberL2ContainingHistoricRoot,
+    uint256 blockNumberL2ContainingHistoricRoot2,
+    Transaction[] memory transactionsOfblockL2ContainingHistoricRoot,
+    Transaction[] memory transactionsOfblockL2ContainingHistoricRoot2,
+    bytes32 salt
+  ) external {
+    checkCommit(msg.data, salt);
+    state.isBlockReal(blockL2, transactions, blockNumberL2);
+    state.isBlockReal(
+      blockL2ContainingHistoricRoot,
+      transactionsOfblockL2ContainingHistoricRoot,
+      blockNumberL2ContainingHistoricRoot
+    );
+    state.isBlockReal(
+      blockL2ContainingHistoricRoot2,
+      transactionsOfblockL2ContainingHistoricRoot2,
+      blockNumberL2ContainingHistoricRoot2
+    );
+    // check the historic root is in the block provided.
```

Comment:
It might be missing this?
```
    require(
      transactions[transactionIndex].historicRootBlockNumberL2[1] ==
      blockNumberL2ContainingHistoricRoot2,
      'Incorrect historic root block'
    );
```

Reply:
yep that's a good catch!